hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java

```diff
@@ -369,6 +384,12 @@ public static Long getMinReplicatedIndex(
         .min(Long::compareTo).orElse(null);
   }
 
+  private static boolean datanodeUseHostName() {
+    return CONF.getBoolean(
+            DFSConfigKeys.DFS_DATANODE_USE_DN_HOSTNAME,
+            DFSConfigKeys.DFS_DATANODE_USE_DN_HOSTNAME_DEFAULT);
```

Comment:
I think we should have an Ozone-specific configuration property for this.  See also:

https://github.com/apache/ozone/blob/ed470082e650a2ab9ac269e70ad737923bae6cb4/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/DFSConfigKeysLegacy.java#L19-L26

Reply:
There are several config properties with prefix `hdds.datanode`.

On a related note, may this setting be applicable to other components in the future?  Is it possible for SCM or OM IP address to change in Kubernetes environment?  If so, we can use a global config instead of one for each component.

Reply:
Yeah. It is possible for SCM and OM to change the IP address. But the solution for them may be different (they may already use the host name in ratis). I need to to take a further look. Lets just make this config for datanode for now.

Reply:
I found that these two properties are used in somewhere else in the code (not my code), that's why I reuse them here. Do you want me to also change property of that place, or you think it is ok we have both hdds.datanode.use.datanode.hostname and dfs.datanode.use.datanode.hostname? @adoroszlai

Reply:
On second thought I think we can stick to `dfs.datanode.use.datanode.hostname` for this PR.  Since there are already usages and there are multiple such legacy keys, I think it's best to deprecate all of them in a separate PR.